app/models/mixins/miq_web_server_worker_mixin.rb

```diff
@@ -87,8 +87,7 @@ def self.sync_workers
             _log.info("Reserved port=#{port}, Current ports in use: #{ports.inspect}")
             ports << port
             ports_hash[:adds] << port
-            w = start_worker(:uri => build_uri(port))
-            result[:adds] << w.pid
+            result[:adds] << Proc.new { start_worker(:uri => build_uri(port)) }
```

Comment:
I think if I don't re-use adds as described above, I'll be able to simplify this and still keep the existing adds to be an array of pids.  I'll see how much clearer that looks or if I should change `result` more as I contemplate also doing the deletes.